superset-frontend/src/dashboard/components/gridComponents/ChartHolder.tsx

```diff
@@ -0,0 +1,333 @@
+import DeleteComponentButton from 'src/dashboard/components/DeleteComponentButton';
+import DragDroppable from 'src/dashboard/components/dnd/DragDroppable';
+import HoverMenu from 'src/dashboard/components/menu/HoverMenu';
+import ResizableContainer from 'src/dashboard/components/resizable/ResizableContainer';
+import getChartAndLabelComponentIdFromPath from 'src/dashboard/util/getChartAndLabelComponentIdFromPath';
+import useFilterFocusHighlightStyles from 'src/dashboard/util/useFilterFocusHighlightStyles';
+import { COLUMN_TYPE, ROW_TYPE } from 'src/dashboard/util/componentTypes';
+import {
+  GRID_BASE_UNIT,
+  GRID_GUTTER_SIZE,
+  GRID_MIN_COLUMN_COUNT,
+  GRID_MIN_ROW_UNITS,
+} from 'src/dashboard/util/constants';
+
+export const CHART_MARGIN = 32;
+
+interface ChartHolderProps {
+  id: string;
+  parentId: string;
+  dashboardId: number;
+  component: LayoutItem;
+  parentComponent: LayoutItem;
+  getComponentById?: (id?: string) => LayoutItem | undefined;
+  index: number;
+  depth: number;
+  editMode: boolean;
+  directPathLastUpdated?: number;
+  fullSizeChartId: number | null;
+  isComponentVisible: boolean;
+
+  // grid related
+  availableColumnCount: number;
+  columnWidth: number;
+  onResizeStart: ResizeStartCallback;
+  onResize: ResizeCallback;
+  onResizeStop: ResizeCallback;
+
+  // dnd
+  deleteComponent: (id: string, parentId: string) => void;
+  updateComponents: Function;
+  handleComponentDrop: (...args: unknown[]) => unknown;
+  setFullSizeChartId: (chartId: number | null) => void;
+  postAddSliceFromDashboard?: () => void;
+}
+
+const ChartHolder: React.FC<ChartHolderProps> = ({
+  id,
+  parentId,
+  component,
+  parentComponent,
+  index,
+  depth,
+  availableColumnCount,
+  columnWidth,
+  onResizeStart,
+  onResize,
+  onResizeStop,
+  editMode,
+  isComponentVisible,
+  dashboardId,
+  fullSizeChartId,
+  getComponentById = () => undefined,
+  deleteComponent,
+  updateComponents,
+  handleComponentDrop,
+  setFullSizeChartId,
+  postAddSliceFromDashboard,
+}) => {
+  const { chartId } = component.meta;
+  const isFullSize = fullSizeChartId === chartId;
+
+  const focusHighlightStyles = useFilterFocusHighlightStyles(chartId);
+  const dashboardState = useSelector(
+    (state: RootState) => state.dashboardState,
+  );
+  const [extraControls, setExtraControls] = useState<Record<string, unknown>>(
+    {},
+  );
+  const [outlinedComponentId, setOutlinedComponentId] = useState<string>();
+  const [outlinedColumnName, setOutlinedColumnName] = useState<string>();
+  const [currentDirectPathLastUpdated, setCurrentDirectPathLastUpdated] =
+    useState(0);
+
+  const directPathToChild = useMemo(
+    () => dashboardState?.directPathToChild ?? [],
+    [dashboardState],
+  );
+
+  const directPathLastUpdated = useMemo(
+    () => dashboardState?.directPathLastUpdated ?? 0,
+    [dashboardState],
+  );
+
+  const infoFromPath = useMemo(
+    () => getChartAndLabelComponentIdFromPath(directPathToChild) as any,
```

Comment:
Is it possible to not use `any` here?

Reply:
We could, but we'd need to change the file that contains that function to typescript, and there's a couple of types around the filters that are not well defined.
Plus, changing that file should follow with tests as well, and I didn't want to make this PR bigger, since it can be addressed as a follow up